### ISSUE TITLE
Some CLI features require x11

### ISSUE THREAD

**Issue by phrack** [bitwarden/directory-connector#17]
This seems like a bug. On a headless Ubuntu 18.04 server if I try to run certain commands I get the following output:

"Cannot autolaunch D-Bus without X11 $DISPLAY"

Specifically I got this when trying to run `sync` and `test`. This is just a test install so I went ahead and install a GUI to be able to complete what I was testing. It turns out that I didn't have a username/password configured. It seems however this message is displayed (perhaps the way it's set to be bold and red?) requires x11. 

This makes it hard to impossible to use the CLI feature on headless systems.

**Comment by kspearrin** [comment]
We write messages to the terminal window using https://github.com/chalk/chalk , which adds color. Maybe that is causing the problem? Is it not possible to write color to the terminal on a headless machine?

**Comment by kspearrin** [comment]
Looks like it is related to libsecret, which we use for storing directory secrets: https://stackoverflow.com/questions/52181225/git-with-libsecret-throws-cannot-autolaunch-d-bus-without-x11-display

**Comment by robin-wittler** [comment]
I installed libsecret but the Error remained. I could work around it with using xvfb but then gnome-keyring-daemon has some problems together with dbus. That was the point where I stopped and decided to take a look to other password management tools.

I think you should write a secure cli tool which does not need a running X-Server Session or interactive input. I want to run the sync job within a docker container - without any human interaction needed.

**Comment by ggiesen** [comment]
Have the same issue on CentOS 7

**Comment by darkyat** [comment]
I have stumbled upon the same issue trying to dockerize the application using ubuntu/debian images. For me the solution was to install, export and launch dbus, as well as manually start and unlock the gnome-keyring-daemon afterwards:

```
apt-get update
apt-get install -y libsecret-1-0 dbus-x11 gnome-keyring
export $(dbus-launch)
dbus-launch
gnome-keyring-daemon --start --daemonize --components=secrets
echo '<RANDOM-PASSPHRASE>' | gnome-keyring-daemon -r -d --unlock
```

Hope this helps!

**Comment by espoelstra** [comment]
You may be able to work around this using the suggestions from the Python keyring library.

https://keyring.readthedocs.io/en/latest/#using-keyring-on-headless-linux-systems

**Comment by CodeCrafter912** [comment]
Any updates on this?
Will there be a simpler way anytime soon?

**Comment by pduchnovsky** [comment]
I personally kind of work around this issue with this:

```
bwdc login $bwlogin $bwpassword
BITWARDENCLI_CONNECTOR_PLAINTEXT_SECRETS=true bwdc sync
bwdc logout
```

Yes it's a bit of overhead but what can we do.
It would be really awesome if a cli version would not be dependent on x11 tbh..

**Comment by dorianim** [comment]
This is still an issue.

**Comment by phrack** [comment]
Agreed that this is still an issue. I'd love to clear the label as the bot suggests, but I don't have the necessary privileges.

**Comment by bert2002** [comment]
This is still a major issue.

**Comment by dkozakRTB** [comment]
confirm in 2025 it is still an issue

**Comment by adityapratap12** [comment]
Hi 
Is there any solution for this? 
`/usr/bin/gnome-keyring-daemon: Operation not permitted`

**Comment by patschi** [comment]
Somewhat shockingly same issue in 2025.11.0. But error is a bit different:
```text
$ bwdc test
Initializing lowdb storage service.
db file "./bitwarden-connector-appdata/data.json already exists"; using existing db
acquiring db file lock
Attempting to create lowdb storage adapter.
Successfully created lowdb storage adapter.
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
acquiring db file lock
Could not connect: No such file or directory
```

When setting `BITWARDENCLI_CONNECTOR_PLAINTEXT_SECRETS=true` it does proceed.
